### PR TITLE
ADD: App Lock && Shake Device to Lock

### PR DESCRIPTION
**Pull request by marcosrdz**
Screenshot_1595299450
Screenshot_1595299452

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/0rvy8pq25hr2xqmkcu1z1zfe14

[download apk](http://filestorage.bluewallet.io:1488/1595300953.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/uvht2z5hphgrbc0xnjv8ntg3x0

[download apk](http://filestorage.bluewallet.io:1488/1595303339.apk)

**Comment by dafacto** [comment]
This PR does not address the issue reported in [Issue 1317](https://github.com/BlueWallet/BlueWallet/issues/1317). The whole point of issue 1317 is that there's something visible in the UI that informs the user that they need to lock the app before leaving; otherwise, the very next opening of the app (possibly under duress) would display the last-open wallet, thereby defeating the whole point of plausible deniability. 

Shake to lock is no more clear to the user as the previous force-quit app.

File: screen/settings/encryptStorage.js [hunk 1]
```diff
@@ -189,6 +202,14 @@ const EncryptStorage = () => {
             Component={TouchableOpacity}
           />
         )}
+        {isShakeToLockedCapable && (
+          <BlueListItemHooks
+            hideChevron
+            title="Shake Device to Lock"
```
[the remark this answers is not in the document]
Comment (dafacto) [reply screen/settings/encryptStorage.js:208]: Don't know if this has been considered, but as part of this change, you would need to replace the modal password prompt, with a lock screen that has a password input. That would be shown on app launch, and after the user manually locks (by shaking or tapping some control.)

**Pull request closed without merging**